### PR TITLE
Separate localSettings file for dev environment

### PR DESCRIPTION
**Pull request by kvas-damian**
It's time to separate ```localSettings``` for dev environment.

**Action required for devs:**
In order to use dev ```localSettings``` you need to update your ```localSettings.ts``` file to target new settings file.

Ping @rogatty @bkoval @rafalkalinski

### CODE REVIEW

**Comment by bkoval** [comment]
LGTM :package: :santa: :jack_o_lantern:

File: server/facets/operations/prepareArticleData.ts [hunk 1]
```diff
@@ -58,10 +58,13 @@ function prepareArticleData (request: Hapi.Request, result: any): void {
 	}
 
 	result.weppyConfig = localSettings.weppy;
+
 	if (typeof result.queryParams.buckysampling === 'number') {
 		result.weppyConfig.samplingRate = result.queryParams.buckysampling / 100;
 	}
 
+	result.localSettings = localSettings;
```
Comment (rogatty) [inline server/facets/operations/prepareArticleData.ts]: Are we sure that there are no secrets in `localSettings`? There is something like `ironSecret` but I don't see it being used anywhere. Until this change we were explicitly saying what should be passed so this is a somewhat breaking change.

Also if we decide that this is fine then you could change things like `result.weppyConfig = localSettings.weppy;` few lines above.

Comment (kvas-damian) [inline server/facets/operations/prepareArticleData.ts]: I don't know if secrets are stored in ```localSettings``` now, but I believe we'll add them there in future. On the other hand passing this variable to view file doesn't mean it's public.

I'm going to cleanup ```weppyConfig``` when we agree that it's fine ;)

File: config/localSettings.base.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ var localSettings: LocalSettings = {
 	// Targeted environment [prod|preview|verify|dev|testing]
 	environment: Utils.getEnvironment(process.env.WIKIA_ENVIRONMENT),
 	helios: {
-		host: process.env.HELIOS_HOST,
+		host: 'https://services.wikia.com/auth',
```
Comment (hakubo) [inline config/localSettings.base.ts]: should this be 
``` 
host: servicesDomain + '/auth'
```
or
```
path: '/auth'
```
and be calculated in run time?

**Comment by hakubo** [comment]
BTW, should we rename those files to 'config'
I don't like the 'local' name in them as they are being used on all different machines now.

File: server/views/_partials/open-graph-tags.hbs [hunk 3]
```diff
@@ -10,6 +10,6 @@
 {{#if openGraph.image}}
 	<meta property="og:image" content="{{openGraph.image}}" />
 {{/if}}
-{{#if wiki.facebookAppId}}
```
Comment (hakubo) [inline server/views/_partials/open-graph-tags.hbs]: should we remove it from API response then?

File: config/localSettings.dev.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+var localSettings = baseLocalSettings.getSettings({
+	loggers: {
+		default: 'debug'
+	},
+	authCookieDomain: '.wikia-dev.com',
+	servicesDomain:'services.wikia-dev.com',
+	helios: {
+		host: 'https://services.wikia-dev.com/auth',
+	},
+	facebook: {
```
Comment (hakubo) [inline config/localSettings.dev.ts]: We should think how can we share config easily between app and mercury

I think we'll have to rewrite some stuff in config repo to json
and create dependency in mercury on config repo

Comment (hakubo) [inline config/localSettings.dev.ts]: this can be done as a separate PR though

**Comment by hakubo** [comment]
Thanks, that make sense.

Ideally we'll move to consul to store configuration for mercury as well

File: server/views/_partials/open-graph-tags.hbs (hunk 3 above)
Comment (kvas-damian) [inline server/views/_partials/open-graph-tags.hbs]: It is: https://github.com/Wikia/app/issues/8126

**Comment by kvas-damian** [comment]
Having those settings in consul makes it almost impossible to start development by contributors that do not have access to our consul. There is the same issue with storing it in chef. 

While all those settings are already public (visible on frontend) I think we could created shared public repo for both app and mercury projects where we could store settings that are common. Then load them in both projects. I agree this should be handled by separate PR.

File: server/views/_partials/open-graph-tags.hbs (hunk 3 above)
Comment (hakubo) [inline server/views/_partials/open-graph-tags.hbs]: ACK :)

**Comment by kvas-damian** [comment]
According to: https://github.com/Wikia/mercury/pull/1236#issuecomment-132215122
I agree, but I'd like to avoid to many changes at once.

File: server/views/_partials/open-graph-tags.hbs (hunk 3 above)
Comment (kvas-damian) [inline server/views/_partials/open-graph-tags.hbs]: FIN ;)

**Comment by lizlux** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

File: server/facets/operations/prepareMainPageData.ts [hunk 5]
```diff
@@ -60,9 +60,11 @@ function prepareMainPageData (request: Hapi.Request, result: any): void {
 		}
 	}
 
-	result.weppyConfig = localSettings.weppy;
-	if (typeof result.queryParams.buckySampling === 'number') {
-		result.weppyConfig.samplingRate = result.queryParams.buckySampling / 100;
+	// clone object to avoid overriding real localSettings for futurue requests
+	result.localSettings = JSON.parse(JSON.stringify(localSettings));
```
Comment (rogatty) [inline server/facets/operations/prepareMainPageData.ts]: Is it better to use `JSON` or `deepExtend` maybe?

**Comment by rogatty** [comment]
:+1:

**Pull request merged**